### DISCUSSION TITLE
How to upgrade to the new version of SecretFlow, SecretPad and Kusia in the prod envs?

### DISCUSSION THREAD

**Discussion by john8628** [secretflow/secretflow#1078] [Q&A] [answered]
How to upgrade to the new version of SecretFlow, SecretPad, and Kusia in the pro env?

**Comment by gshilei** [comment, marked as answer]
Hi @john8628 

当前在生产环境使用Secretpad, Kuscia, Secretflow的一些注意事项：

secretpad:
- 默认方式部署时，Secretpad使用的是本地Sqlite数据库存放平台中的信息，生产环境推荐替换为外部的Mysql存储(当前Secretpad暂未验证使用Mysql存储，接下来我们会内部验证下)，实现数据持久化。如果想将当前本地的Secretpad升级为使用Mysql存储，那么需要将Sqlite中的数据迁移至Mysql存储中。

Kuscia:
- 默认方式部署时，Kuscia Master/Kuscia Autonomy使用的是本地Sqlite数据库存放集群中的信息，生产环境推荐替换为外部Mysql存储，实现数据持久化。

Secretflow:
- 由于Secretpad还未开放应用镜像多版本管理功能，因此当前Secretpad不支持多版本Secretflow镜像的选择。也就是说仅能使用升级后的Secretflow镜像运行作业。 可参考注册隐语镜像章节(https://www.secretflow.org.cn/docs/quickstart/mfbgum8vi3ngs4y9#Wxq6s)更新Secretpad平台组件列表和在Kuscia中注册自定义算法镜像。

此外，Secretpad/Kuscia正在开发迭代中，后面我们也会在官方文档中逐步完善生产环境的推荐配置。不同版本之间的镜像升级，可能会有一些不兼容性。针对不兼容的升级，未来我们会在文档中补充版本升级的详细步骤。

**Reply by john8628** [reply]
ok, thanks for your reply
